src/modules/game/engine/systems/hop.ts

```diff
@@ -23,12 +25,12 @@ export default class HopSystem extends System {
 		let needRefresh = false
 		this.hopQuery.added.forEach((entity) => {
 			const hop = entity.c[Hop.key] as Hop
-			const actorCell = entity.c[MapCell.key].cell
-			const target = actorCell.getHopTarget(hop)
-			if (target) {
-				actorCell.reserveTarget(target)
-				actorCell.properties.platform = false
-				Object.assign(hop, target)
+			const actorCell = entity.c[MapCell.key].cell as Cell3D
+			const validHop = getValidHop(actorCell, hop)
+			if (validHop) {
+				Object.assign(hop, validHop)
```

Comment:
Expose a method here too.

Reply:
Adding a setGrid method to the Hop component is an option, but in this case we actually only need to update the `z` property so I've changed it to do that.